plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2294,8 +2307,9 @@ private void cleanExtraOutputFiles(ConnectorSession session, Set<String> written
     {
         TrinoFileSystem fileSystem = fileSystemFactory.create(session);
         Set<String> locations = getOutputFilesLocations(writtenFiles);
+        Set<String> fileNames = getOutputFilesFileNames(writtenFiles);
         for (String location : locations) {
-            cleanExtraOutputFiles(fileSystem, session.getQueryId(), location, writtenFiles);
+            cleanExtraOutputFiles(fileSystem, session.getQueryId(), location, fileNames);
```

Comment:
Existing style thing, but the two `cleanExtraOutputFiles` should probably be next to each other in the file